brave/brave-core
# Local state pref to use from Java

Resolves https://github.com/brave/brave-browser/issues/27632

Intruduced BraveLocalState class.

Removed methods from BravePrefServiceBridge which were used to access these local_state prefs:
   - BravePref.P3A_ENABLED
   - BravePref.P3A_NOTICE_ACKNOWLEDGED
   - BravePref.STATS_REPORTING_ENABLED
   - BravePref.SAFETYNET_STATUS
   - BravePref.ENS_RESOLVE_METHOD
   - BravePref.SNS_RESOLVE_METHOD
   - BravePref.UNSTOPPABLE_DOMAINS_RESOLVE_METHOD
   - BravePref.ENS_OFFCHAIN_RESOLVE_METHOD

Removed unused Java/JNI methods:
   - BravePrefServiceBridge.setReferralAndroidFirstRunTimestamp;
   - BravePrefServiceBridge.setReferralCheckedForPromoCodeFile;
   - BravePrefServiceBridge.setReferralInitialization;
   - BravePrefServiceBridge.setReferralPromoCode;
   - BravePrefServiceBridge.setReferralDownloadId.


## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
New features are not added with this PR. The test plan is required to verify nothing is broken.

I. `brave.p3a.enabled`
1. Make fresh installation
2. Launch browser, on FTRW uncheck `Share completely private and anonymous product insights about what features are being used by Brave's users.`
3. Open brave://local-state, must exist
```
 "p3a": {
         "enabled": true,
```
entry.
4. Go to app menu => Settings => Brave Shields & privacy, `Allow privacy-preserving product diagnostic (P3A)` must be unchecked
5. Check `Allow privacy-preserving product diagnostic (P3A)` switch
6. Open brave://local-state, the entry should look
```
 "p3a": {
         "enabled": false,
```


II. `brave.stats.reporting_enabled`
1. Open app menu => Settings => Brave Shields & privacy, turn off `Automatically send daily usage pings` 
2. Open brave://local-state, find `reporting_enabled` should be false
3. Open again app menu => Settings => Brave Shields & privacy, `Automatically send daily usage pings` must be unchecked


III. `safetynet.status`
According to the code, the Brave Rewards grant should be retreived successfullt to verify it works as before.


IV. `brave.unstoppable_domains.resolve_method`
1. Open app menu => Settings => Brave Shields & privacy => Unstoppable Domains, switch it to `Enabled` 
2. Open brave://local-state, find `resolve_method` near `unstoppable_domains`, it should be
```
"unstoppable_domains": {
         "resolve_method": 3
      },
```
3. Relaunch browser, open app menu => Settings => Brave Shields & privacy => Unstoppable Domains, it should still be `Enabled`

V. `brave.ens.resolve_method`
1. Open app menu => Settings => Brave Shields & privacy => Ethereum Name Service, switch to `Enabled`
2. Open brave://local-state, find `resolve_method` near `ens`, it should be
```
"ens": {
         "resolve_method": 3
      },
```
3. Relaunch browser, open app menu => Settings => Brave Shields & privacy => Ethereum Name Service, it should still be `Enabled`

VI. `brave.sns.resolve_method`
1. Open brave://flags => `Enable Solana Name Service support` => Enabled, relaunch browser
2. Open app menu => Settings => Brave Shields & privacy => Solana Name Service, switch to `Enabled`
3. Open brave://local-state, find `resolve_method` near `sns`, it should be
```
"sns": {
         "resolve_method": 3
      },
```
4. Relaunch browser, open app menu => Settings => Brave Shields & privacy => Solana Name Service, it should still be `Enabled`


VII. `brave.ens.offchain_resolve_method`
1. Open app menu => Settings => Brave Shields & privacy => Unstoppable Domains, switch `ENS offchain lookup` to `Enabled` 
2. Open brave://local-state, find `resolve_method` near `unstoppable_domains`, it should be
```
"ens": {
         "offchain_resolve_method": 2,
```
3. Relaunch browser, open app menu => Settings => Brave Shields & privacy => Unstoppable Domains, `ENS offchain lookup` should still be `Enabled`
